### ISSUE TITLE
All twitters tab refresh on add-on install

### ISSUE THREAD

**Issue by HYBRID-BEING** [ZusorCode/GoodTwitter#14]
That was quite unexpected. I've got tons of tabs and a lot of them are twitter ones, most are not even loaded at the moment (bad practice, i know, but that's what i got used to). Needless to say that was quite heavy on my Firefox.
I understand this is by design, but maybe you can give a prompt before refreshing each and every one of those tabs and maybe even allow user to opt-out of this massive refresh (but warn about cached data)?

**Comment by HYBRID-BEING** [comment]
They also seem to refresh on add-on update, which can happen at any moment. Heck, i've been reading my twitter list when the tab suddenly refreshed. Please add a warning at least.

**Comment by ZusorCode** [comment]
The add-on needs to refresh to activate the changes as sometimes twitter messes with it if it isn't reloaded. I don't know how to only make it refresh on a fresh install. Closed unless someone knows of a solution for this

**Comment by HYBRID-BEING** [comment]
What about the prompt/warning though?

**Comment by ZusorCode** [comment]
I can't create a pop-up without injecting it into a webpage.

**Comment by HYBRID-BEING** [comment]
Um, just to double-check - is it not possible to avoid refreshing **unloaded** tabs? It also looks like GoodTwitter affects Firefox startup time greatly, so i'm planning on uninstalling it, to my dismay (I only need GT for "See x new tweets" button/counter in lists, and it's still not brought back).

**Comment by ZusorCode** [comment]
I haven't figured out a way to see if a tab is loaded. It should also only reload tabs on install and updates (I haven't pushed an update in 2 months). As for the startup, I've never experienced anything like that. What version of Firefox/GoodTwitter are you on?

**Comment by ZusorCode** [comment]
It shouldn't cause anything to slow down the browser on startup. The only reason why that might be is if you have a bunch of Twitter tabs open and your browser reopens them on startup. They way GoodTwitter works is it replaces the User Agent with an old browser so it needs to inspect requests and replace the user agent but that only happens on Twitter tabs

**Comment by HYBRID-BEING** [comment]
As for reloading tabs - i think it also happened when browser got updated (i have to admit though, i wasn't actually using Firefox, but it's fork Waterfox). I've updated it yesterday (to the latest version - 2019.10) and all my twitter tabs loaded on startup (they don't usually load by themselves). I've tried restarting again and though it was slow, so i tried disabling add-on and for few times it seemed to work, but not anymore, so, yeah, slow startup is most likely not related to GT after all.
Still, what about an option to not refresh tabs automatically, off by default and with a huge warning about requiring force refresh via Ctrl+F5?
